pkg/local_object_storage/metabase/exists.go

```diff
@@ -100,6 +100,11 @@ func (db *DB) exists(tx *bbolt.Tx, addr oid.Address, currEpoch uint64) (exists b
 //   - 2 if object is covered with tombstone;
 //   - 3 if object is expired.
 func objectStatus(tx *bbolt.Tx, addr oid.Address, currEpoch uint64) uint8 {
+	// locked object could not be removed/marked with GC/expired
+	if objectLocked(tx, addr.Container(), addr.Object()) {
+		return 0
+	}
+
```

Comment:
Can we have an explicit unit-test for this behaviour?

Reply:
sure, added